test/extensions/filters/http/oauth2/filter_test.cc

```diff
@@ -1237,6 +1364,327 @@ TEST_F(OAuth2Test, OAuthBearerTokenFlowFromQueryParameters) {
             filter_->decodeHeaders(request_headers, false));
 }
 
+TEST_F(OAuth2Test, OAuthTestFullFlowWithUseRefreshToken) {
```

Comment:
Maybe add a description to this test talking about what this tests - seems like it first does one authorization that doesn't use the refresh token followed by one that does?